src/Microsoft.ML.Data/DataLoadSave/CompositeDataLoader.cs

```diff
@@ -1,616 +1,118 @@
-        private readonly TransformEx[] _transforms;
-        private readonly ITransposeDataView _tview;
-        private readonly IHost _host;
-
         /// <summary>
-        /// Returns the underlying data view of the composite loader.
-        /// This can be used to programmatically explore the chain of transforms that's inside the composite loader.
+        /// The underlying data loader.
         /// </summary>
-        public IDataView View { get; }
-
+        public readonly IDataLoader<TSource> Loader;
         /// <summary>
-        /// Creates a loader according to the specified <paramref name="args"/>.
-        /// If there are transforms, then the result will be a <see cref="CompositeDataLoader"/>,
-        /// otherwise, it'll be whatever <see cref="IDataLoader"/> is specified in <c>args.loader</c>.
+        /// The chain of transformers (possibly empty) that are applied to data upon loading.
         /// </summary>
-        public static IDataLoader Create(IHostEnvironment env, Arguments args, IMultiStreamSource files)
-        {
-            Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
-
-            h.CheckValue(args, nameof(args));
-            h.CheckValue(args.Loader, nameof(args.Loader));
-            h.CheckValue(files, nameof(files));
+        public readonly TransformerChain<TLastTransformer> Transformer;
 
-            var loader = args.Loader.CreateComponent(h, files);
-            return CreateCore(h, loader, args.Transforms);
-        }
-
-        /// <summary>
-        /// Creates a <see cref="CompositeDataLoader"/> that starts with the <paramref name="srcLoader"/>,
-        /// and follows with transforms created from the <paramref name="transformArgs"/> array.
-        /// If there are no transforms, the <paramref name="srcLoader"/> is returned.
-        /// </summary>
-        public static IDataLoader Create(IHostEnvironment env, IDataLoader srcLoader,
-            params KeyValuePair<string, IComponentFactory<IDataView, IDataTransform>>[] transformArgs)
+        public CompositeDataLoader(IDataLoader<TSource> loader, TransformerChain<TLastTransformer> transformerChain = null)
         {
-            Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
+            Contracts.CheckValue(loader, nameof(loader));
+            Contracts.CheckValueOrNull(transformerChain);
 
-            h.CheckValue(srcLoader, nameof(srcLoader));
-            h.CheckValueOrNull(transformArgs);
-            return CreateCore(h, srcLoader, transformArgs);
+            Loader = loader;
+            Transformer = transformerChain ?? new TransformerChain<TLastTransformer>();
         }
 
-        private static IDataLoader CreateCore(IHost host, IDataLoader srcLoader,
-            KeyValuePair<string, IComponentFactory<IDataView, IDataTransform>>[] transformArgs)
+        public IDataView Load(TSource input)
```

Comment:
>Load [](start = 25, length = 4)

Add a line of comment similar to the IDataLoader.Load that explains this is lazy.